### PR TITLE
Linear Regression in online mode was added

### PR DESCRIPTION
**Pull request by prutskov**
# Description
Linear Regression in Online computation mode with examples were added.

### CODE REVIEW

**Comment by prutskov** [comment]
PCm: http://intel-ci.intel.com/ea3ce5f0-015d-f148-9d40-8cdcd4b723a1

**Comment by prutskov** [comment]
PCm[2]: http://intel-ci.intel.com/ea3ea2d1-937f-f1fb-8765-8cdcd4b723a1

**Comment by prutskov** [comment]
PCm[3]: http://intel-ci.intel.com/ea41a0ec-742a-f13e-add7-8cdcd4b723a1

**Comment by prutskov** [comment]
PCm[4]: http://intel-ci.intel.com/ea41db6b-00cf-f1d3-80c0-8cdcd4b723a1

File: algorithms/kernel/linear_regression/oneapi/linear_regression_train_kernel_oneapi.h [hunk 1]
```diff
@@ -68,6 +67,22 @@ class BatchKernelOneAPI<algorithmFPType, training::normEqDense> : public daal::a
                              bool interceptFlag) const;
 };
 
+template <typename algorithmFPType, training::Method method>
+class OnlineKernelOneAPI
+{};
+
+template <typename algorithmFPType>
+class OnlineKernelOneAPI<algorithmFPType, training::normEqDense> : public daal::algorithms::Kernel
```
Comment (amgrigoriev) [inline algorithms/kernel/linear_regression/oneapi/linear_regression_train_kernel_oneapi.h]: In some other algorithms, kernel name includes brief algorithm name. Is it in accordance with our naming policy?

Comment (prutskov) [reply algorithms/kernel/linear_regression/oneapi/linear_regression_train_kernel_oneapi.h]: This kernel is placed in namespace of algorithm. With this approach collisions will not occur.

File: algorithms/kernel/linear_regression/oneapi/linear_regression_train_dense_normeq_oneapi_impl.i [hunk 2]
```diff
@@ -47,6 +46,21 @@ services::Status BatchKernelOneAPI<algorithmFPType, training::normEqDense>::comp
     return status;
 }
 
+template <typename algorithmFPType>
+services::Status OnlineKernelOneAPI<algorithmFPType, training::normEqDense>::compute(NumericTable & x, NumericTable & y, NumericTable & xtx,
+                                                                          NumericTable & xty, bool interceptFlag) const
```
[the remark this answers is not in the document]
Comment (prutskov) [reply algorithms/kernel/linear_regression/oneapi/linear_regression_train_dense_normeq_oneapi_impl.i]: fixed

File: algorithms/kernel/linear_regression/linear_regression_train_container.h [hunk 3]
```diff
@@ -127,7 +127,17 @@ Status BatchContainer<algorithmFPType, method, cpu>::compute()
 template <typename algorithmFPType, training::Method method, CpuType cpu>
 OnlineContainer<algorithmFPType, method, cpu>::OnlineContainer(Environment::env * daalEnv)
 {
-    __DAAL_INITIALIZE_KERNELS(internal::OnlineKernel, algorithmFPType, method);
+    auto & context    = oneapi::internal::getDefaultContext();
+    auto & deviceInfo = context.getInfoDevice();
```
[the remark this answers is not in the document]
Comment (prutskov) [reply algorithms/kernel/linear_regression/linear_regression_train_container.h]: Context iface has not const method "deviceInfo" and we cann't use const in this case.

**Comment by prutskov** [comment]
@michael-smirnov, Result allocation using sycl tables are presented in files `\algorithms\kernel\linear_model\linear_model_model_fpt.cpp` and `\algorithms\kernel\linear_regression\linear_regression_ne_model_fpt.cpp`

**Comment by prutskov** [comment]
PCm[5]: http://intel-ci.intel.com/ea429767-ede4-f128-a3a0-8cdcd4b723a1

**Comment by prutskov** [comment]
PCm[6]: http://intel-ci.intel.com/ea4322dd-f9dc-f1e4-a5e0-8cdcd4b723a1

**Pull request merged**